src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -112,9 +112,23 @@ public int Count
         
         public override int GetHashCode()
         {
-            return null == _array
-                        ? 0
-                        : _array.GetHashCode() ^ _offset ^ _count;
+            if (_array == null)
+            {
+                return 0;
+            }
+            
+#if !FEATURE_CORECLR
+            // Old implementation: Xor everything.
+            // This isn't optimal since it treats _offset and _count interchangeably,
+            // but we keep it around in the full framework for compat.
+            return _array.GetHashCode() ^ _offset ^ _count;
+#else
+            // Taken from ValueTuple.CombineHashCodes
+            int hash = _array.GetHashCode();
+            hash = ((hash << 5) + hash) ^ _offset;
+            hash = ((hash << 5) + hash) ^ _count;
```

Comment:
@JonHanna I'm choosing to stick with `33` for now; I've done a bit of Googling on the subject, and it looks like although the latter is composite, 31 and 33 both result in very few collisions. See here: http://math.stackexchange.com/a/59952 and here: https://gist.github.com/hmic/1676398#file-hash-h-L20-L21 As 33 seems to be more idiomatic when calculating Bernstein hashes, I'm sticking with how it's implemented as-is.

One practical advantage of using `31` for hash calculations instead is that it can be [slightly cheaper to calculate on RISC](http://stackoverflow.com/a/35304979/4077294), but I'm not too familiar with the code generation-aspect of CoreCLR, so I don't know if changing this to `(hash << 5) - hash` would lead to any real perf benefits on those platforms.